### PR TITLE
Port back feature from nobleo-noetic-devel where we also report the message that made stuff go bad

### PR DESCRIPTION
**Pull request by MCFurry**
Just like we did in ROS1, we also want to report the message that made the diagnostics go bad, to toplevel state.
This is also since some of our Harvey algorithms depend on this single conclusion

### CODE REVIEW

**Comment by Timple** [comment]
I'll review later. But I'd like PR 324 to be merged before we open a new PR with these changes.

**Comment by MCFurry** [comment]
Sure thing, but our nobleo:ros2 branch is already different from our PR variant, so merging this won't hurt that PR?

File: diagnostic_aggregator/src/aggregator.cpp [hunk 1]
```diff
@@ -229,27 +231,57 @@ void Aggregator::publishData()
...
+    diag_toplevel_state.message = msg_to_report->message;
+    diag_toplevel_state.hardware_id = msg_to_report->hardware_id;
+    diag_toplevel_state.values = msg_to_report->values;
+  }
 
   std::vector<std::shared_ptr<DiagnosticStatus>> processed_other =
     other_analyzer_->report();
   for (const auto & msg : processed_other) {
     diag_array.status.push_back(*msg);
+    const uint depth = static_cast<uint>(std::count(msg->name.begin(), msg->name.end(), '/'));
```
Comment (Timple) [inline diagnostic_aggregator/src/aggregator.cpp]:
```c++
const auto depth = std::count(msg->name.begin(), msg->name.end(), '/');
```
Why the cast?
`non_ok_status_depth` can have the same type in this case.

**Comment by Timple** [comment]
Would it be possible to base of the `ros2` branch instead of `ros2-nobleo`? This PR shouldn't change, but it'll allow us to easily open a PR later. Or squash all commits into one.

**Comment by Timple** [comment]
I can't find the button :/
But you have my approval ;-)

**Pull request merged**